### PR TITLE
Add sections on Intel ME dependency

### PR DESCRIPTION
**Pull request by plbossart**

### CODE REVIEW

File: getting_started/intel_debug/suggestions.rst [hunk 1]
```diff
@@ -27,6 +27,19 @@ enable two of the four speakers present. All of these cases are orthogonal
...
+
+If the ME is disabled by the OEM or the user, firmware authentication
+will fail. In case of boot errors, the user should try the following
+command
+
+.. code-block::
+
+   ls /dev/mei*/
+
+If no devices are reported, then the user needs to enable the ME.
```
Comment (marc-hb) [inline getting_started/intel_debug/suggestions.rst:41]: As noted by @fredoh9 , maybe mention here that `/dev/mei*/` requires some Kconfig in the first place.

Also mention that firmware authentication itself does NOT require any `/dev/mei*/`, which could be confusing. Calling it a "debug interface" should avoid any confusion even though it's not just a debug interface.

**Comment by plbossart** [comment]
please don't merge for now, the part on /dev/mei is likely to require a significant update.

**Review by lgirdwood — COMMENTED** [review]
This is useful info.

**Comment by plbossart** [comment]
updated (no way to detect ME enablement) and added section on HDaudio amps.

File: getting_started/intel_debug/introduction.rst [hunk 2]
```diff
@@ -104,6 +104,18 @@ installing their own firmware. Notable exceptions include Google
 Chromebooks and Up2/Up-Extreme boards, where the 'community key' is
 used.
 
+The Intel ME (Management Engine) is responsible for authentication of
+the firmware, be it signed by an Intel production key, a community key
+or an OEM key. If the Intel ME is disabled by an OEM, or disabled by
```
Comment (marc-hb) [inline getting_started/intel_debug/introduction.rst:109]: ..., be it signed by an Intel or OEM production key (consumer products) or by a community key (development systems and make boards)

File: getting_started/intel_debug/suggestions.rst [hunk 3]
```diff
@@ -27,6 +27,30 @@ enable two of the four speakers present. All of these cases are orthogonal
...
+reverse-engineer the missing configuration by snooping HDaudio
+commands in a Windows environment.
+
+Make sure the ME is enabled
+***************************
+
+If the ME is disabled by the OEM or the user, firmware authentication
+will fail without an explicit log provided to the user. In case of any
+authentication failure, the user should verify that the ME is not
+disabled.
```
Comment (marc-hb) [inline getting_started/intel_debug/suggestions.rst:52]: ..., more information about the ME in the "firmware binary" section above.

File: getting_started/intel_debug/suggestions.rst [hunk 4]
```diff
@@ -27,6 +27,30 @@ enable two of the four speakers present. All of these cases are orthogonal
...
+and are not SOF bugs proper. To fix such issues, it is necessary to
+either get direct support from the codec vendor, or alternatively to
+reverse-engineer the missing configuration by snooping HDaudio
+commands in a Windows environment.
+
+Make sure the ME is enabled
+***************************
+
+If the ME is disabled by the OEM or the user, firmware authentication
+will fail without an explicit log provided to the user. In case of any
```
Comment (marc-hb) [inline getting_started/intel_debug/suggestions.rst:50]: without any explicit feedback provided to the user

(more generic)

File: getting_started/intel_debug/introduction.rst (hunk 2 above)
Comment (lgirdwood) [reply getting_started/intel_debug/introduction.rst:109]: Chromebooks also use community key.

**Review by lgirdwood — APPROVED** [review]
@deb-intel are you able to fix up the grammar after merge as you usually do ? @marc-hb has pointed out a few grammar changes., but there maybe others. Need to merge for v1.7

**Comment by plbossart** [comment]
pushed changes suggested by @marc-hb

**Review by marc-hb — APPROVED** [review]

**Review by deb-intel — APPROVED** [review]
Will make grammatical updates in a separate PR.

**Pull request merged**